Created dot.md in the numpy/concepts/built-in-functions/terms/dot (#4610)
Codecademy/docs: content/numpy/concepts/built-in-functions/terms/dot/dot.md

---
Title: '.dot()'
Description: 'Computes the dot product of two arrays.' 
Subjects: 
  - 'Computer Science'
  - 'Data Science'
Tags: 
  - 'Arrays'
  - 'Data Structures'
  - 'Functions'
  - 'Linear Algebra'
  - 'NumPy'
CatalogContent: 
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
  - 'paths/data-science-foundations'  
---

> The **`np.dot()`** function computes the dot product of two arrays `a` and `b`, with the result varying based on the input. The function returns an [`ndarray`](https://www.codecademy.com/resources/docs/numpy/ndarray), and if both `a` and `b` are either scalars or 1-D arrays, a scalar is returned. Otherwise, an array is returned.

Reviewer: ```suggestion
In NumPy, the **`.dot()`** method computes the dot product of two arrays. For 1-D arrays, it provides a scalar value. When both arrays are 2-D, it performs matrix multiplication, resulting in a new 2-D array. Further details about various scenarios, including arrays with more than two dimensions, can be found in the NumPy [`.dot()` documentation](https://numpy.org/doc/stable/reference/generated/numpy.dot.html). This method is widely utilized in linear algebra and numerical computations.
```

## Syntax

```pseudo
numpy.dot(a, b, out=None)
```

> - `a`: The first array (or scalar) to be multiplied.
> - `b`: The second array (or scalar) to be multiplied.
> - `out`: Optional parameter that specifies the output array where the result is stored. If not provided, a new array is created.

Reviewer: ```suggestion
- `a`: The first array, which could be 1-D, 2-D, or higher dimensional.
- `b`: The second array, having compatible dimensions with `a`, and also could be 1-D, 2-D, or higher dimensional.
- `out`: Optional parameter that specifies the output array where the result is stored. If not provided, a new array is created.
```

## Example

> 

Reviewer: ```suggestion

The usage of the .dot() method is demonstrated in the following example:

```

```py
import numpy as np

# defining 2 arrays
array_a = [3,4,5]
array_b = [1,2,3]

# computing and printing the product
print(np.dot(array_a, array_b))
```

This produces the following output:

```shell
26
```

## Codebyte Example

```codebyte/python
import numpy as np

m = [[3, 5], [1, 4], [5, 4]] 
n = [[7, 3, 1], [2, 7, 2]] 
print("Matrix m :") 
print(m) 
print("Matrix n :") 
print(n) 
  
result = np.dot(m, n) 
  

> print("The matrix multiplication is :") 

Reviewer: ```suggestion
print("The dot product of matrix m and n is :") 
```

> ---
> Title: '.dot()'
> 
> Description: 'Dot product of two arrays.' 
> 
> Subjects: 
>   - 'Data Science'
>   - 'Computer Science'
> 
> Tags: 
>   - 'Data Structures'
>   - 'Arrays'
>   - 'Linear Algebra'
>   - 'Functions'
>   - 'NumPy'
> 
> CatalogContent: 
>   - 'learn-python-3'
>   - 'paths/computer-science'
>   - 'paths/data-science'
>   - 'paths/data-science-foundations'  
> ---

Reviewer 2: ```suggestion
---
Title: '.dot()'
Description: 'Computes the dot product of two arrays.' 
Subjects: 
  - 'Computer Science'
  - 'Data Science'
Tags: 
  - 'Arrays'
  - 'Data Structures'
  - 'Functions'
  - 'Linear Algebra'
  - 'NumPy'
CatalogContent: 
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
  - 'paths/data-science-foundations'  
---
```

Reviewer 2: Please accept the above suggestion; there should not be any line spacing in the meta data and it should be alphabetically arranged.

The **`np.dot()`** function computes the dot product of two arrays `a` and `b`, with the result varying based on the input. The function returns an [`ndarray`](https://www.codecademy.com/resources/docs/numpy/ndarray), and if both `a` and `b` are either scalars or 1-D arrays, a scalar is returned. Otherwise, an array is returned.

## Syntax

```pseudo
numpy.dot(a, b, out=None)
```

> Parameters:
> - 'a: array_like, first argument'
> - 'b: array_like, second argument'
> - 'out: ndarray, optional'

Reviewer 2: ```suggestion

- `a`: The first array (or scalar) to be multiplied.
- `b`: The second array (or scalar) to be multiplied.
- `out`: Optional parameter that specifies the output array where the result is stored. If not provided, a new array is created.
```

Reviewer 2: Please accept the above suggestion, since it describes the parameters in depth.

> Returns:
> - output:ndarray 
> 

Reviewer 2: ```suggestion
```

Reviewer 2: Please remove the Return block, it is not needed here.

## Example

```py
import numpy as np

# defining 2 arrays
array_a = [3,4,5]
array_b = [1,2,3]

# computing and printing the product
print(np.dot(array_a, array_b))
```

> ```shell
> 26
> ```

Reviewer 2: ```suggestion

```shell
26
```
```

Reviewer 2: Please accept the suggestion, a one-line spacing was needed.

## Codebyte Example

> ```py

Reviewer 2: ```suggestion

```codebyte/python
```

import numpy as np

> # creating two matrices 

Reviewer 2: ```suggestion
```

m = [[3, 5], [1, 4], [5, 4]] 
n = [[7, 3, 1], [2, 7, 2]] 
print("Matrix m :") 
print(m) 
print("Matrix n :") 
print(n) 
  

> # computing product 

Reviewer 2: ```suggestion
```

result = np.dot(m, n) 
  

> # printing the result 

Reviewer 2: ```suggestion
```

Reviewer 2: Please accept the above code byte suggestions, since comments are not required for a codebyte.

Generally the function/operation accepts 3 types of inputs: 
1. 1D arrays [3,6,8,...]
2. 2D arrays [4,5,6]
             [7,8,9]
             [1,2,3]
3. scalar = a real number, usually but not always positive (simply put it is an int or a float).

Math Examples:
scalar * scalar: np.dot(2,3.25) = 6.5
scalar * 1D array: np.dot(4, [2,4,6]) = array([8,16,24])
1D array (or Python List) * 1D array: np.dot([1,2,3],[4,5,6]) = (1x4)+(2x5)+(3x6)= 32
2D array * 2D array uses matrix multiplication. Here is a simple example:
Matrix Multiplication: A = [3,4,5] B = [10,11]
                           [6,7,8]     [12,13]
                                       [14,15]
A gives us 2 rows, B gives us 2 columns final out put will be a 2X2 matrix:[a,b]
                                                                           [c,d]
a = (3x10)+(4x12)+(5x14)=148, b = (3x11)+(4x13)+(5x15)=160
c = (6x10)+(7x12)+(8x14)=256, b = (6x11)+(7x13)+(8x15)=277
output = [148,160]
         [256,277]

> 

Reviewer 2: This can be removed. Line 26 to 47

## Syntax

```pseudo

> numpy.dot(a, b, *(optional)out=None*)

Reviewer 2: ```suggestion
numpy.dot(a, b, out=None)
```

Reviewer 2: After the syntax use the unordererd list format (bullet) to describe these parameters as:

- `a`: Parameter description
- `b`: Parameter description
- `out`: Parameter description

```

## Example

```py
import numpy as np

# defining 2 arrays
array_a = [3,4,5]
array_b = [1,2,3]

# computing and printing the product
print(np.dot(array_a, array_b))
```

> output: 26

Reviewer 2: ```suggestion

```shell
26
```
```

## Example 2 Matrix Multiplication

```py
import numpy as np

# creating two matrices 
m = [[3, 5], [1, 4], [5, 4]] 
n = [[7, 3, 1], [2, 7, 2]] 
print("Matrix m :") 
print(m) 
print("Matrix n :") 
print(n) 
  
# computing product 
result = np.dot(m, n) 
  
# printing the result 
print("The matrix multiplication is :") 
print(result) 
```
output:
Matrix m :
[[3, 5], [1, 4], [5, 4]]
Matrix n :
[[7, 3, 1], [2, 7, 2]]
The matrix multiplication is :
[[31 44 13]
 [15 31  9]

>  [43 43 13]]

Reviewer 2: Instead of Example 2, place it in the ## Codebyte Example block. And, for the output, use the ```shell``` block. Refer to the [Style Guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md).

Author: @mamtawardhani Thank you for your assistance with this terms project. I have made the changes you suggested. I will be performing another PR for further review.

Author: It looks like I don't need to perform another PR; however, please review the changes. Let me know if more edits are needed.
